lib/kb_ObjectInfo/kb_ObjectInfoImpl.py

```diff
@@ -518,21 +541,71 @@ def genomeset_report(self, ctx, params):
         multi_fasta = []
         rpt_string = ''
         
+#       Always do the list of genomes - short and ensures that there is an HTML link
+        gsr = CreateMultiGenomeReport(self.config)
+        rpt_list = gsr.readGenomeSet(genome_name, genomeset_data)
+        htmltable = self.make_HTML(rpt_list,'col_header')
+        html_report_path = os.path.join(self.scratch, 'genomeset_comparison.html')
+        html_report_txt = open(html_report_path, "w")
+        html_report_txt.write("<h1>GENOME COMPARISON</h1>")
+        html_report_txt.write(htmltable)
+        html_report_txt.close()
+        
         if params['showGenomes']:
-            gsr = CreateMultiGenomeReport(self.config)
-            rpt_list = gsr.readGenomeSet(genome_name, genomeset_data)
             rpt_string += self.write_to_file(rpt_list,'genomeset_tab_file.tsv',"\t")
             self.write_to_file(rpt_list,'genomeset_cvs_file.csv',",")
-            htmltable = self.make_HTML(rpt_list,'col_header')
-            
-            html_report_path = os.path.join(self.scratch, 'genomeset_comparison.html')
+
+        if params['listCoding']:
+            cf = CreateFeatureLists(self.config)
+            myGS = genomeset_data['elements']
+            rpt_list = []
+            for ele in myGS:
+                genome = self.dfu.get_objects({'object_refs': [myGS[ele]['ref']]})
+                genome_data = genome['data'][0]['data']
+
+                rpt_list += cf.delimitedTable(genome_data, 'features')
+                
+            if rpt_list:
+                rpt_string += self.write_to_file(rpt_list,'genome_tab_file.tsv',"\t")
+                self.write_to_file(rpt_list,'genome_csv_file.csv',",")
+                
+            html_report_path = os.path.join(self.scratch, 'genome_features.html')
             html_report_txt = open(html_report_path, "w")
-            html_report_txt.write("<h1>GENOME COMPARISON</h1>")
+            htmltable = self.make_HTML(rpt_list,'col_header')
+            html_report_txt.write("<h1>PROTEIN CODING FEATURES</h1>")
             html_report_txt.write(htmltable)
             html_report_txt.close()
-        
+                
+        if params['listGFF']:
+            myGS = genomeset_data['elements']
```

Comment:
This would have been good advise to have during the first review of the module when I was overhauling everything. For this PR I'm going to focus on the revisions being made to the genomeSet objects.

Reply:
Agreed. It isn't really a big deal. The fact that there are issues like this to bring up is a sign that everything is in the code is so healthy!
Really, I was just taking the place of the code quality tools we have in other repos which will generate alerts for this kind of thing. I don't know how picky other reviewers are.